src/async_traits.rs

```diff
@@ -262,20 +263,26 @@ where
                         //
                         // We're still in the transaction, but we at least
                         // tried to ROLLBACK to our savepoint.
-                        if !retryable_error(&err) || !retry().await {
+                        let retried = match &err {
+                            RunError::User(err) => retryable_error(err) && retry().await,
+                            RunError::RuntimeShutdown => false,
+                        };
```

Comment:
wondered briefly if `retryable_error` ought to just take a `RunError`, but it looks like it's only called in two places, and one of them is a place where we've already extracted the user error, so...nevermind.

Reply:
Yeah